api-registration/server/index.js

```diff
@@ -0,0 +1,54 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  argon2
+    .hash(password)
+    .then(hashedPassword => {
+      const sql = `
+          insert into "users" ("username", "hashedPassword")
+          values ($1, $2)
+          returning "username", "userId", "createdAt"
+        `;
+
+      const params = [username, hashedPassword];
+      db.query(sql, params)
+        .then(result => {
+          res.status(201).json(result.rows);
```

Comment:
This will send back an array with the user object in it, we want to send just the object, the first item in the array.
```JAVASCRIPT
res.status(201).json(result.rows[0]);
```

Reply:
<img width="743" alt="image" src="https://user-images.githubusercontent.com/46618757/156691334-f17bf06b-3db1-44f1-9da1-55c332b8ed96.png">